google/cloud/spanner_dbapi/connection.py

```diff
@@ -37,11 +34,46 @@ class Connection:
     """
 
     def __init__(self, instance, database):
+        self._pool = BurstyPool()
```

Comment:
Having a connection pool for each `Connection` is too much. One connection will never use more than one session at any time, as:
1. A connection can execute at most one transaction at any time.
1. A session can execute at most one (read/write) transaction at any time.

This is fine for a prototype, for production purposes, the implementation should do one of the following:
1. Create one session pool globally that is reused for all connections that are opened. This is the preferable implementation, as the connections would benefit from all optimizations that might be done to the session pool implementation.
1. OR: Open one session (not session pool) for each connection, and use that session for all operations on that connection. Conceptually, a Cloud Spanner session is quite comparable to a Python database connection. Implementing it this way would however mean that the connection needs to do all session management by itself, including things like keep-alive pings, handling the backend dropping a session, etc. It is therefore not recommended to do that, and instead rely on the session pool handling all that for you.